modules/Rubrics/moduleFunctions.php

```diff
@@ -240,128 +242,244 @@ function rubricView($guid, $connection2, $gibbonRubricID, $mark, $gibbonPersonID
+                                $visualise = true ;
+                            }
                         }
 
-                        $col->addContent($historicalContent)
-                            ->addClass('historical')
-                            ->prepend('<b><u>' . __('Total Occurences:') . ' ' . $countHistorical . '</u></b><br/>')
-                            ->append(($countHistorical > 7)? '<b>'.__('Older occurrences not shown...').'</b>' : '')
-                            ->append('<span class="cellID" data-cell="' . $cell['gibbonRubricCellID'] . '"></span>');
+                        if ($visualise) {
+                            foreach ($contexts as $entry) {
+                                if ($entry['gibbonRubricCellID'] == $cell['gibbonRubricCellID']) {
+                                    $means[$row['gibbonRubricRowID']]['cumulative'] += $cellCount;
+                                    $means[$row['gibbonRubricRowID']]['denonimator'] ++; 
+                                }
+                            }
+                            $cellCount++;
+                        }
                     }
                 }
-            }
 
-            if ($mark == true) {
-                $output .= "<script type='text/javascript'>";
-                $output .= '$(document).ready(function(){';
-                $output .= '$(".markableCell").parent().click(function(){';
-                    $output .= "var mode = '';";
-                    $output .= "var cellID = $(this).find('.cellID').data('cell');";
-                    $output .= "if ($(this).hasClass('rubricCellHighlight') == false ) {";
-                        $output .= "$(this).addClass('rubricCellHighlight');";
-                        $output .= "mode = 'Add';";
-                    $output .= '} else {';
-                        $output .= "$(this).removeClass('rubricCellHighlight');";
-                        $output .= "mode = 'Remove';";
-                    $output .= '}';
-                    $output .= 'var request=$.ajax({ url: "'.$_SESSION[$guid]['absoluteURL'].'/modules/Rubrics/rubrics_data_saveAjax.php", type: "GET", data: {mode: mode, gibbonRubricID : "' . $gibbonRubricID.'", gibbonPersonID : "'.$gibbonPersonID.'", gibbonRubricCellID : cellID, contextDBTable : "'.$contextDBTable.'",contextDBTableID : "'.$contextDBTableID.'"}, dataType: "html"});';
-                    $output .= '});';
-                $output .= '});';
-                $output .= '</script>';
-            }
+                $output .= "<div id='canvasHolder' style='width:100%;'>";
+                    $output .= "<canvas id='birdsCanvas'></canvas>";
+                $output .= "</div>";
+
+                $labels = "" ;
+                foreach ($means as $mean) {
+                    $labels .= "\"".$mean['title']."\",";
+                }
+                if ($labels != '') {
+                    $labels = substr($labels, 0, -1);
```

Comment:
This is a somewhat clunky way to concatenate csv strings 😉 An alternate might be to collect them as an array and then implode with a `,`

Reply:
I've had a go at this, and it is now simpler. Could it be more elegant? Perhaps!